File: sdkwork-backend-sdk-go/api/feeds.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type FeedsApi struct {
    client *sdkhttp.Client
}

func NewFeedsApi(client *sdkhttp.Client) *FeedsApi {
    return &FeedsApi{client: client}
}

// Update an existing feeds entry
func (a *FeedsApi) Update(body sdktypes.PlusFeedsForm) (sdktypes.PlusApiResultPlusFeedsVO, error) {
    raw, err := a.client.Put(BackendApiPath("/feeds"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusFeedsVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusFeedsVO](raw)
}

// Create a new feeds entry
func (a *FeedsApi) Create(body sdktypes.PlusFeedsForm) (sdktypes.PlusApiResultPlusFeedsVO, error) {
    raw, err := a.client.Post(BackendApiPath("/feeds"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusFeedsVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusFeedsVO](raw)
}

// Get feeds entries by page
func (a *FeedsApi) ListByPage(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusFeedsVO, error) {
    raw, err := a.client.Post(BackendApiPath("/feeds/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusFeedsVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusFeedsVO](raw)
}

// Get all feeds entries
func (a *FeedsApi) ListAllEntities(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusFeedsVO, error) {
    raw, err := a.client.Post(BackendApiPath("/feeds/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusFeedsVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusFeedsVO](raw)
}

// Get a feeds entry by ID
func (a *FeedsApi) GetById(id string) (sdktypes.PlusApiResultPlusFeedsVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/feeds/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusFeedsVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusFeedsVO](raw)
}

// Delete a feeds entry
func (a *FeedsApi) Delete(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/feeds/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

File: sdkwork-backend-sdk-typescript/src/api/feeds.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultBoolean, PlusApiResultListPlusFeedsVO, PlusApiResultPagePlusFeedsVO, PlusApiResultPlusFeedsVO, PlusFeedsForm, QueryListForm } from '../types';


export class FeedsApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Update an existing feeds entry */
  async update(body: PlusFeedsForm): Promise<PlusApiResultPlusFeedsVO> {
    return this.client.put<PlusApiResultPlusFeedsVO>(backendApiPath(`/feeds`), body);
  }

/** Create a new feeds entry */
  async create(body: PlusFeedsForm): Promise<PlusApiResultPlusFeedsVO> {
    return this.client.post<PlusApiResultPlusFeedsVO>(backendApiPath(`/feeds`), body);
  }

/** Get feeds entries by page */
  async listByPage(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusFeedsVO> {
    return this.client.post<PlusApiResultPagePlusFeedsVO>(backendApiPath(`/feeds/list`), body, params);
  }

/** Get all feeds entries */
  async listAllEntities(body?: QueryListForm): Promise<PlusApiResultListPlusFeedsVO> {
    return this.client.post<PlusApiResultListPlusFeedsVO>(backendApiPath(`/feeds/list/all`), body);
  }

/** Get a feeds entry by ID */
  async getById(id: string | number): Promise<PlusApiResultPlusFeedsVO> {
    return this.client.get<PlusApiResultPlusFeedsVO>(backendApiPath(`/feeds/${id}`));
  }

/** Delete a feeds entry */
  async delete(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/feeds/${id}`));
  }
}

export function createFeedsApi(client: HttpClient): FeedsApi {
  return new FeedsApi(client);
}
